test/suspend.js

```diff
@@ -0,0 +1,68 @@
+import {serial as test} from 'ava';
+import sinon from 'sinon';
+
+const mockery = require('mockery');
+
+const logUpdateApi = {
+	main(text) {
+		console.log(text);
+	}
+};
+logUpdateApi.main.clear = function () {
+};
+
+logUpdateApi.main.done = function () {
+};
+
+const mock = sinon.mock(logUpdateApi);
+
+function getTasks() {
+	// Require is necessary at this position, otherwise mockery is not working correctly
+	const Listr = require('..');
+	const tasks = new Listr([
+		{
+			title: 'not suspending task',
+			task: () => {
+				return new Promise(resolve => {
+					setTimeout(() => {
+						resolve('predefined output');
+					}, 150);
+				});
+			}
+		},
+		{
+			title: 'suspending task',
+			task: () => {
+				return new Promise(resolve => {
+					setTimeout(() => {
+						resolve('predefined output');
+					}, 4000);
+				});
+			},
+			options: {suspendUpdateRenderer: true}
+		}
+	]);
+
+	return tasks;
+}
+
+test.before(() => {
+	/* One time for the first task,
+	   one time when both tasks are finished
+	*/
+	mock.expects('main').twice();
+	mockery.registerAllowable('..');
+	mockery.registerMock('log-update', logUpdateApi.main);
+	mockery.enable({useCleanCache: true, warnOnUnregistered: false});
+});
+
+test('should suspend second task', async t => {
```

Comment:
This doesn't test use-cases where `suspendUpdateRenderer` is actually needed. There should be a test(s) for that as well.

Reply:
(See https://github.com/SamVerschueren/listr-update-renderer/pull/23#issuecomment-531460048)

Reply:
My intention was to create two task. The first task without the option `suspendUpdateRenderer`. So the update renderer should display status "task 1 pending" exactly one time. The second task with the option `suspendUpdateRenderer`. So the update renderer shouldn't display status "task 2 pending", because it is suspended. If the suspend-function doesn't work as expected, the update renderer displays "task 2 pending" more than once, because we have a timeout of 4s. 
You could simply try it, when cloning my fork `listr` locally. With the current version of listr-update-renderer, the tests fail. With linking my fork of `listr-update-renderer` the tests pass.

What tests you thought of? Integration-tests, which ask for user input or tests, which mock `listr-update-renderer`?

Reply:
What I thought of is a test case in which the suspension is actually required, meaning that the test won't pass if the task isn't suspended (and only passes if it is).

Reply:
I added now a custom renderer mock, which output is excatly verified. It only passes, if suspension is activated.
When the mock isn't called with the correct args, the test fails.

Reply:
I'm sorry to keep bothering you, but this is still not what I meant in my comments above.
As I recall, the reason we need this in np is since one of the commands we call waits for the user to enter a password, which we need to receive before moving on to the next task.
This test, again, only verifies we correctly pass on the `suspendUpdateRenderer` option from [here](https://github.com/SamVerschueren/listr/pull/136/files#diff-b538cfad262b8bd2b69eb3ba895dc505R59) to the Listr renderer (let me know if I'm mistaken, though).
I think that the simplest way to test this would be to actually run a command that waits for user input, then verify we do receive it when `suspendUpdateRenderer` is set to `true` (and don't when it's set to `false`, as the task isn't suspended).

Reply:
The problem isn't, that the task can't receive the user-input from `stdin`. The task actually receives the input made to `stdin`. The problem is, that `listr-update-renderer` erases password-prompts on `stdout`. For the user it looks like the prompt never appears and the task is running infinite.

Reply:
@itaisteinherz
Btw i added the testclass `task-with-input`, which proves, that the input from `stdin` is available for every task, no just for the tasks with suspended renderer.